torch/onnx/_internal/fx/torch_export_graph_extractor.py

```diff
@@ -42,6 +42,9 @@ def generate_fx(
         #         args=model_args,  # type: ignore[arg-type]
         #         kwargs=model_kwargs,  # type: ignore[arg-type]
         #     )
+        # TODO: input_adapter and output_adapter should be based on pytree flatten/unflatten API?
+
+        model = model.run_decompositions(options.decomposition_table)
```

Comment:
Put it in `pre_export_passes`?

Reply:
This is tricky. Although we have implemented decomposition as a `GraphModule` pass for the `torch._dynamo.export` `fx_tracer`, `torch.export` does decomp at the `ExportedProgram` level. Through the `torch.export` + `ExportedProgram` flow, we run decompositions before we start doing passes of the `GraphModule` (using `self.pre_export_passes`)

In fact, at `generate_fx` we call `return self.pre_export_passes()` passing `original_model=model` and `fx_module=model.graph_module`. Note that we extract `GraphModule` from the `ExportedProgram.graph_module` after the decomposition was ran. Only then we start doing `GraphModule` passes (and ignore `original_model`).

In order to move the `ExportedProgram` decomposition to `pre_export_passes` we have to 
1) change `model = model.run_decompositions(options.decomposition_table)` call (which assigns a `ExportedProgram` to another ExportedProgram`) to `fx_module = original_model.run_decompositions(options.decomposition_table).graph_module` (which assigns a `GraphModule` coming from a `ExportedProgram`; that is the `generate_fx`'s role, not `pre_export_passes`)
2) Always run `fx_module: GraphModule = exported_program.run_decompositions` as the first pass and 
3) Remove `fx_module` as the input to the `pre_export_passes`, meaning `generate_fx` is not generating the `GraphModule` anymore

Although techinically possible, it mixes responsibilities of `generate_fx` and `pre_export_passes`.

Reply:
Okay. We are missing out diagnostics for decomp though.

Reply:
how should we address this?

Reply:
Need to think more on that. Good opportunity to explore how to track low level info such as these from other modules.